### PR TITLE
[Feature][Task Plugin] Support hive cli task plugin

### LINKED ISSUE #11261: [Feature][Task Plugin] Support hive cli task type
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

* Currently DS only supports running hive sql through jdbc. We could add hive cli task plugin to enable users to run hive sql in client mode for stability in prod environment.
* In some companies, pure `shell` tasks are banned for security reasons. It will be better to provide such a `hive Cli` task.

### Use case

* Already described above.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by EricGao888**
## Purpose of the pull request

* Add `Hive Cli` task plugin
* Currently DS only supports running hive sql through jdbc. This PR adds hive cli task plugin to enable users to run hive sql using hive client for better stability in prod environment.
* In some companies, pure `shell` tasks are banned for security reasons. It will be better to provide such a `hive Cli` task.
* This PR closes: #11261 

## Brief change log

* Already described above.
* Will add docs in the following commits.

## Verify this pull request

* Verified by UTs and manual tests.

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-hivecli/src/main/java/org/apache/dolphinscheduler/plugin/task/hivecli/HiveCliConstants.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+@UtilityClass
+public class HiveCliConstants {
+
+    public static final String TYPE_SCRIPT = "SCRIPT";
+
+    public static final String TYPE_FILE = "FILE";
+
+    public static final String HIVE_CLI_EXECUTE_FILE = "hive -f";
+
+    public static final String HIVE_CLI_EXECUTE_SCRIPT = "hive -e \"%s\"";
```
Comment (SbloodyS) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-hivecli/src/main/java/org/apache/dolphinscheduler/plugin/task/hivecli/HiveCliConstants.java:31]: I think we can provide `-v` and `--hiveconf` as options for users to choose.

**Comment by zhuxt2015** [comment]
What if hive with kerberos? Most companies will enable kerberos in the production environment.

**Comment by EricGao888** [comment]
> What if hive with kerberos? Most companies will enable kerberos in the production environment.

@zhuxt2015 Good question. AFAIK, currently DS task plugins do not support `kerberos` except `SQL tasks with datasource`. I've added instructions in docs about how to choose between `hive datasource` and `hive cli task plugin` for different scenarios. If they enable `kerberos`, they could use `SQL tasks with hive datasource` to connect to `hiveserver2` with `jdbc` at this moment.

However, it will be great if other task plugins like `spark task plugin`, `hive cli task plugin`, etc. could support `kerberos` in the future. May I ask whether you have any suggestions or ideas on how to do it? Thanks.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-hivecli/src/main/java/org/apache/dolphinscheduler/plugin/task/hivecli/HiveCliConstants.java (hunk 1 above)
Comment (EricGao888) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-hivecli/src/main/java/org/apache/dolphinscheduler/plugin/task/hivecli/HiveCliConstants.java:31]: done

**Comment by EricGao888** [comment]
BTW, we might combine `spark task plugin` with `spark datasource` and `hive cli task plugin` with `hive datasource` when doing #10283 so that users could manage similar tasks in the same plugin.

**Review by SbloodyS — DISMISSED** [review]
LGTM

File: dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-hive-cli.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import * as Fields from '../fields/index'
+import type { IJsonItem, INodeData, ITaskData } from '../types'
+
+export function useHiveCli({
+                               projectCode,
+                               from = 0,
+                               readonly,
+                               data
+                           }: {
+    projectCode: number
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-hive-cli.ts]: It's better to format the code by eslint.

File: dolphinscheduler-ui/src/locales/zh_CN/project.ts [hunk 3]
```diff
@@ -644,6 +644,12 @@ export default {
     zeppelin_parameters_tips: '请输入zeppelin dynamic form参数',
     zeppelin_rest_endpoint: 'zeppelinRestEndpoint',
     zeppelin_rest_endpoint_tips: '请输入zeppelin server的rest endpoint',
+    hive_cli_task_execution_type: 'Hive Cli 任务类型',
+    hive_cli_task_execution_type_tips: '请选择您的Hive Cli任务类型',
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/locales/zh_CN/project.ts]: This seems not to be used, same as the en_US.

File: dolphinscheduler-ui/src/locales/zh_CN/project.ts [hunk 4]
```diff
@@ -644,6 +644,12 @@ export default {
...
+    hive_sql_script: 'Hive SQL 脚本',
+    hive_sql_script_tips: '请填写您的Hive SQL脚本',
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/locales/zh_CN/project.ts]: This seems not to be used too.

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-hive-cli.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            props: {
+                language: 'sql'
+            }
+        },
+        {
+            type: 'input',
+            field: 'hiveCliOptions',
+            name: t('project.node.hive_cli_options'),
+            props: {
+                placeholder: t('project.node.hive_cli_options')
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-hive-cli.ts]: It's better to use the 'hive_cli_options_tips' here.

File: dolphinscheduler-ui/src/views/projects/task/components/node/format-data.ts [hunk 6]
```diff
@@ -658,5 +664,10 @@ export function formatModel(data: ITaskData) {
   if (data.taskParams?.jobType) {
     params.isCustomTask = data.taskParams.jobType === 'CUSTOM'
   }
+
+  if (data.taskParams?.type) {
+    params.type = data.taskParams.type
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/format-data.ts]: It seems that the type doesn't need to be reassigned here.

Comment (EricGao888) [reply dolphinscheduler-ui/src/views/projects/task/components/node/format-data.ts]: Sure I will remove it.

File: dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-hive-cli.ts (hunk 5 above)
Comment (EricGao888) [reply dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-hive-cli.ts]: Sure, will fix it in the next commit.

File: dolphinscheduler-ui/src/locales/zh_CN/project.ts (hunk 4 above)
Comment (EricGao888) [reply dolphinscheduler-ui/src/locales/zh_CN/project.ts]: You mean `hive_sql_script_tips`? I will remove it in the next commit.

File: dolphinscheduler-ui/src/locales/zh_CN/project.ts (hunk 3 above)
Comment (EricGao888) [reply dolphinscheduler-ui/src/locales/zh_CN/project.ts]: Sure will remove it

File: dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-hive-cli.ts (hunk 2 above)
Comment (EricGao888) [reply dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-hive-cli.ts]: May I ask whether how I could format it with `eslint`? Is there any command I could use to run?

Comment (Amy0104) [reply dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-hive-cli.ts]: run 'npm run lint'

Comment (EricGao888) [reply dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-hive-cli.ts]: @Amy0104 I ran `npm run lint` under `dolphinscheduler-ui` directory, thus some other files are reformatted as well. PTAL when available, thanks!

Comment (Amy0104) [reply dolphinscheduler-ui/src/views/projects/task/components/node/tasks/use-hive-cli.ts]: We will add a husky to lint code later.

**Review by Amy0104 — DISMISSED** [review]
The front end part LGTM.

**Review by SbloodyS — APPROVED** [review]

**Comment by zhuxt2015** [comment]
> > What if hive with kerberos? Most companies will enable kerberos in the production environment.
> 
> @zhuxt2015 Good question. AFAIK, currently DS task plugins do not support `kerberos` except `SQL tasks with datasource`. I've added instructions in docs about how to choose between `hive datasource` and `hive cli task plugin` for different scenarios. If they enable `kerberos`, they could use `SQL tasks with hive datasource` to connect to `hiveserver2` with `jdbc` at this moment.
> 
> However, it will be great if other task plugins like `spark task plugin`, `hive cli task plugin`, etc. could support `kerberos` in the future. May I ask whether you have any suggestions or ideas on how to do it? Thanks.

1. hive command bypasses permission management,  In most cases it will be disabled in a production environment。
2. If you have to use hive command with kerberos be enabled. execute `kinit -kt <keytab path> <principal> ` before execute hive command ,

**Comment by EricGao888** [comment]
> > > What if hive with kerberos? Most companies will enable kerberos in the production environment.
> > 
> > 
> > @zhuxt2015 Good question. AFAIK, currently DS task plugins do not support `kerberos` except `SQL tasks with datasource`. I've added instructions in docs about how to choose between `hive datasource` and `hive cli task plugin` for different scenarios. If they enable `kerberos`, they could use `SQL tasks with hive datasource` to connect to `hiveserver2` with `jdbc` at this moment.
> > However, it will be great if other task plugins like `spark task plugin`, `hive cli task plugin`, etc. could support `kerberos` in the future. May I ask whether you have any suggestions or ideas on how to do it? Thanks.
> 
> 1. hive command bypasses permission management,  In most cases it will be disabled in a production environment。
> 2. If you have to use hive command with kerberos be enabled. execute `kinit -kt <keytab path> <principal> ` before execute hive command ,

@zhuxt2015 Thanks for the information.

1. For the first point, we have the same situation with `spark-submit`, `spark-sql`, `flink run`, etc. besides `hive` considering permissions. We could abstract this and make it apply to all those task plugins. Would you like to submit an issue and give a shot? @zhuxt2015 
2. `HiveServer2` is mainly used for `ad-hoc` query. In production, `HiveServer2` is not a good practice for scheduled big data tasks. We have seen several cases of production failures.

**Pull request merged**